### ISSUE TITLE
Need some help debugging classPath to add gradle support to coc-groovy

### ISSUE THREAD

**Issue by joshzcold** [dansomething/coc-groovy#11]
Hey there, im trying to add gradle support to coc-groovy and I added the operations to generate the groovy classpath folowing the same convention as you do in maven

https://github.com/joshzcold/coc-groovy/blob/master/src/classpath.ts

generated this file:

`/home/joshua/.config/coc/extensions/coc-groovy-data/groovy_ws_03593d465d7c807b6c208582f2979b90/.groovy-classpath`

contents summary:

```
...
/home/joshua/.gradle/caches/modules-2/files-2.1/org.seleniumhq.selenium/selenium-java/3.141.59/e8e20b741299cb8ffedbd2925a3ab69d08b41063/selenium-java-3.141.59.jar:/home/joshua/.gradle/caches/modules-2/files-2.1/org.codehaus.groovy/groovy-all/2.4.4/b22748
b6cf9a38c9777792175507a679043cdd64/groovy-all-2.4.4.jar
...etc
```
coc-groovy seems to create and sync the file sucessfully.

unfortunatley im still getting unfound errors from groovy-language-server it seems like
![image](https://user-images.githubusercontent.com/36175703/115761250-62d0be00-a35f-11eb-8443-b2ff228272d2.png)

My question is 
1. what would be the best way to debug this
2. does groovy-language-server only need the classpath string[] to resolve the dependencies or am I missing something? Is there anything else coc-groovy would have to send to the server?
3. At this point should I get in contact with the groovy-language-server devs?
4. maybe groovy-language-server is hardcoded to `/home/user/.m2/`?

groovy-language-server supplied jar is from coc-groovy 
launching coc-groovy via `Plug 'joshzcold/coc-groovy', {'do': 'yarn install --frozen-lockfile'}`
thank you.

**Comment by joshzcold** [comment]
I am getting completion for src files. i.e `src.my.class` so groovy-language-server seems to be operational

**Comment by dansomething** [comment]
> 1. what would be the best way to debug this

The first thing you'll want to do is open the Groovy language server trace logs and search for `classpath`. It should include the array of jar file paths that was generated. If it's there the language server will load them as long as it can read them. I know from personal experience that it can take a while for the language server to process all of the jar files on a project with a lot of dependencies. Perhaps you aren't waiting long enough?

> 2. does groovy-language-server only need the classpath string[] to resolve the dependencies or am I missing something? Is there anything else coc-groovy would have to send to the server?

The Groovy language server only looks for `settings.groovy.classpath`. IIRC, each string entry can be a path to a jar or a directory containing jar files.

> 3. At this point should I get in contact with the groovy-language-server devs?

Right now I doubt its an issue with the language server.

> 4. maybe groovy-language-server is hardcoded to /home/user/.m2/?

AFAIK, the language server has no awareness of build tools like Maven or Gradle.

**Comment by joshzcold** [comment]
I got it figured out. 
Since I was experimenting with this before I set my coc settings to 
```json
{
	"groovy":{
		"project":{
			"referencedLibraries":"/home/joshua/.gradle/caches/modules-2/files-2.1/*"
		}	
	}
}
```
so when coc-groovy did `referencedLibraries.concat(..)` it created one long string and sent it to the server where the server was checking for an array. 
`&& groovy.get("classpath").isJsonArray()) {`

Ill have a pull request for gradle support soon, it works pretty good now.
